### PR TITLE
7 transitioning common to typescript

### LINKED ISSUE #7: Transitioning /common to TypeScript
### Description:
Houses reusable components to use in a variety of situations within the app (Loading Spinners, Alerts, User Context, and Signup form)

### Goals:
- [ ] Convert all JavaScript files to TypeScript.
- [ ] Ensure all components, functions, and variables have proper typings.
- [ ] Resolve any TypeScript compilation errors.
- [ ] Update import/export statements to TypeScript syntax.
- [ ] Refactor any necessary code to comply with TypeScript best practices.
- [ ] Ensure TypeScript strict mode is enabled and configured appropriately.

### Steps to Complete:
1. **Analysis:**
   - [ ] Identify all JavaScript files within the directory.
   - [ ] Determine the types of components, functions, and variables used.

2. **Conversion:**
   - [ ] Rename `.jsx`|`.js` files to `.tsx`|`.ts`.
   - [ ] Add TypeScript typings to all components, functions, and variables.

3. **TypeScript Configuration:**
   - [ ] Ensure `tsconfig.json` is properly configured.
   - [ ] Enable TypeScript strict mode (`strict: true`).
   - [ ] Configure any necessary TypeScript options for the directory.

4. **Testing and Debugging:**
   - [ ] Test the application to ensure proper functionality after conversion.
   - [ ] Address and fix any TypeScript compilation errors.
   - [ ] Debug any runtime issues that arise from the transition.

5. **Documentation:**
   - [ ] Document any significant changes made during the transition process.
   - [ ] Update README or relevant documentation with TypeScript usage guidelines for the directory.

### Additional Notes:
[Include any additional notes or considerations relevant to this directory's transition.]

### Project:
[Jobly Transition to TypeScript](https://github.com/users/nathanrydel/projects/5)

### Labels:
- [x] TypeScript
- [x] Transition
- [x] Directory: common

### PR DESCRIPTION
**Pull request by SarthakSanjay**
Fixes #7

### CODE REVIEW

**Comment by evengene** [comment]
hey @SarthakSanjay, thank you for contributing,
Could you please in the future name your branch according to the issue name?  
For example, in this case `7-transitioning-common-to-ts`
This approach keeps PR merge history clean and makes it easy to navigate through them.
thanks!

File: frontend/src/common/Alert.tsx [hunk 1]
```diff
@@ -4,19 +4,23 @@ import React from "react";
  *
  * { LoginForm, SignupForm, ProfileForm } -> Alert
  **/
+export interface AlertProps {
+  type : string,
+  messages: string[]
+}
 
-function Alert({ type = "danger", messages = [] }) {
+const Alert: React.FC<AlertProps> = ({ type = "danger", messages = [] }) => {
```
Comment (evengene) [inline frontend/src/common/Alert.tsx]: you can import `FC` directly from react, like this:
```
import type { FC } from 'react';

const Alert: FC<AlertProps>
...
```

File: frontend/src/common/SearchForm.tsx [hunk 2]
```diff
@@ -12,22 +12,25 @@ import "./SearchForm.css";
  *
  * { CompanyList, JobList } -> SearchForm
  */
+interface SearchFormProps{
```
Comment (evengene) [inline frontend/src/common/SearchForm.tsx]: `...Props { ` - formatting

File: frontend/src/common/SearchForm.tsx [hunk 3]
```diff
@@ -12,22 +12,25 @@ import "./SearchForm.css";
...
+  searchFor: (term: string | undefined) => void;
+}
 
-function SearchForm({ searchFor }) {
+const SearchForm: React.FC<SearchFormProps> = ({ searchFor }) => {
```
Comment (evengene) [inline frontend/src/common/SearchForm.tsx]: same as above, you can import just  `FC`

File: frontend/src/common/SearchForm.tsx [hunk 4]
```diff
@@ -12,22 +12,25 @@ import "./SearchForm.css";
...
   console.debug("SearchForm", "searchFor=", typeof searchFor);
 
-  const [searchTerm, setSearchTerm] = useState("");
+  const [searchTerm, setSearchTerm] = useState<string>("");
 
   /** Tell parent to filter */
-  function handleSubmit(evt) {
+  const handleSubmit = (evt:React.FormEvent):void => {
```
Comment (evengene) [inline frontend/src/common/SearchForm.tsx]: same as above

File: frontend/src/common/SearchForm.tsx [hunk 5]
```diff
@@ -12,22 +12,25 @@ import "./SearchForm.css";
...
     // take care of accidentally trying to search for just spaces
     evt.preventDefault();
     searchFor(searchTerm.trim() || undefined);
     setSearchTerm(searchTerm.trim());
   }
 
   /** Update form fields */
-  function handleChange(evt) {
+  const handleChange = (evt:React.ChangeEvent<HTMLInputElement>):void => {
```
Comment (evengene) [inline frontend/src/common/SearchForm.tsx]: same as above

**Review by evengene — CHANGES_REQUESTED** [review]

**Comment by SarthakSanjay** [comment]
@evengene thanks for the review

**Review by nathanrydel — APPROVED** [review]

**Pull request merged**